core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -264,7 +268,7 @@ final class ShuffleBlockFetcherIterator(
 
       override def onBlockFetchFailure(blockId: String, e: Throwable): Unit = {
         logError(s"Failed to get block(s) from ${req.address.host}:${req.address.port}", e)
-        results.put(new FailureFetchResult(BlockId(blockId), infoMap(blockId)._2, address, e))
+        throwFetchFailedException(BlockId(blockId), infoMap(blockId)._2, address, e)
       }
```

Comment:
`onBlockFetchFailure` will be called within Netty's thread. Thus, `ShuffleBlockFetcherIterator` can not catch the exception.

Reply:
thanks, have modified the code.